neonvm/controllers/virtualmachine_controller.go

```diff
@@ -503,9 +503,68 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 		// TODO: implement RestartPolicy
 		switch virtualmachine.Spec.RestartPolicy {
 		case vmv1.RestartPolicyAlways:
-			// TODO: restart VirtualMachine
+			log.Info("Restarting VM runner pod", "VM.Phase", virtualmachine.Status.Phase, "RestartPolicy", virtualmachine.Spec.RestartPolicy)
+			// get runner to delete
+			vmRunner := &corev1.Pod{}
+			err := r.Get(ctx, types.NamespacedName{Name: virtualmachine.Status.PodName, Namespace: virtualmachine.Namespace}, vmRunner)
+			if err == nil {
+				// delete current runner
+				if err = r.Delete(ctx, vmRunner); err != nil {
+					log.Error(err, "Failed to delete runner Pod", "Pod.Namespace", vmRunner.Namespace, "Pod.Name", vmRunner.Name)
+					return err
+				}
+				r.Recorder.Event(virtualmachine, "Normal", "Deleted",
+					fmt.Sprintf("VM runner pod was deleted: %s", vmRunner.Name))
+				log.Info("VM runner pod was deleted", "Pod.Namespace", vmRunner.Namespace, "Pod.Name", vmRunner.Name)
+				// TODO: remove thos hack after redesign reconcile loop
+				time.Sleep(time.Second)
```

Comment:
are TODOs tracked somewhere (issues, centrally?) on what time scale are we expecting to resolve this? 

also a comment about why 1 second was chosen and what would happen (assuming the refactor happens after we merge this) if we changed this in either direction.

Reply:
`time.Sleep` has been removed as it now works without delay. In my tests with the wrong vm phase cleanup, this resulted in duplicate runner pods, but now everything is fine.